bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxLocalAccessHandler.java

```diff
@@ -90,68 +101,192 @@ private void retrieveData() {
         }
     }
 
-    private void updateData(String rawJsonData) {
+    private void updateFromData(String rawJsonData) {
         try {
-            LocalConnectRawDataBean inverterParsedData = parseJson(rawJsonData);
-            updateThing(inverterParsedData);
+            LocalConnectRawDataBean rawDataBean = parseJson(rawJsonData);
+            InverterType inverterType = calculateInverterType(rawDataBean);
+            RawDataParser parser = inverterType.getParser();
+            if (parser != null) {
+                if (!alreadyRemovedUnsupportedChannels) {
+                    removeUnsupportedChannels(inverterType.getSupportedChannels());
+                    alreadyRemovedUnsupportedChannels = true;
+                }
+
+                updateChannels(rawDataBean, parser);
+
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } else {
+                cancelSchedule();
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Parser for inverter of type " + inverterType.name() + " is not implemented.");
```

Comment:
Can you add an I18N text for this, something like:
```suggestion
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                        "@text/offline.configuration-error.parser-not-implemented [\"" + inverterType.name() + "\"]");
```

with:
```ini
offline.configuration-error.parser-not-implemented = Parser for inverter of type {0} is not implemented.
```